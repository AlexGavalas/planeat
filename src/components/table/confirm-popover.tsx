import { Button, Group, Stack, Text } from '@mantine/core';
import { useTranslation } from 'next-i18next';
import { type MouseEventHandler } from 'react';

type ConfirmationPopoverProps = {
    isDeleteInProgress: boolean;
    onToggleConfirmation: () => void;
    onDelete: MouseEventHandler<HTMLButtonElement>;
};

export const ConfirmationPopover = ({
    onDelete,
    isDeleteInProgress,
    onToggleConfirmation,
}: ConfirmationPopoverProps) => {
    const { t } = useTranslation();

    return (
        <Stack align="center" gap="md">
            <Text>{t('confirmation.generic')}</Text>
            <Group gap="md">
                <Button
                    color="red"
                    onClick={onToggleConfirmation}
                    size="xs"
                    variant="outline"
                >
                    {t('confirmation.no')}
                </Button>
                <Button
                    loading={isDeleteInProgress}
                    onClick={onDelete}
                    size="xs"
                >
                    {t('confirmation.yes')}
                </Button>
            </Group>
        </Stack>
    );
};
