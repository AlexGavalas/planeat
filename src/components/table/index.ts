export { INITIAL_PAGE, PAGE_SIZE, Table } from './table';
