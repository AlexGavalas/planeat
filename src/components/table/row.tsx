import {
    ActionIcon,
    Button,
    Group,
    Popover,
    Stack,
    TableTd,
    TableTr,
    Text,
} from '@mantine/core';
import { EditPencil, Trash } from 'iconoir-react';
import get from 'lodash/fp/get';
import { useTranslation } from 'next-i18next';
import {
    type MouseEventHandler,
    type ReactNode,
    useCallback,
    useState,
} from 'react';

export type Item = Record<string, ReactNode> & {
    id: string | number;
};

export type Header<ItemType> = {
    width: string;
    label: string;
    key: string;
    formatValue?: (item: ItemType) => string;
};

type RowProps<ItemType> = {
    item: ItemType;
    headers: Header<ItemType>[];
    onDelete: (item: ItemType) => Promise<void> | void;
    onEdit: (item: ItemType) => Promise<void> | void;
};

export const Row = <ItemType extends Item>({
    item,
    headers,
    onDelete,
    onEdit,
}: RowProps<ItemType>) => {
    const { t } = useTranslation();
    const [hasOpenConfirmation, setHasOpenConfirmation] = useState(false);
    const [deleteInProgress, setDeleteInProgress] = useState(false);

    const handleDelete = useCallback<
        MouseEventHandler<HTMLButtonElement>
    >(async () => {
        setDeleteInProgress(true);

        await onDelete(item);

        setDeleteInProgress(false);
        setHasOpenConfirmation(false);
    }, [item, onDelete]);

    const handleEdit = useCallback<
        MouseEventHandler<HTMLButtonElement>
    >(async () => {
        await onEdit(item);
    }, [item, onEdit]);

    const toggleConfirmation = useCallback(() => {
        setHasOpenConfirmation(true);
    }, []);

    return (
        <TableTr style={{ width: '100%', height: '3rem' }}>
            {headers
                .filter((header) => header.key !== 'actions')
                .map(({ key, width, formatValue }) => (
                    <TableTd key={key} style={{ width }}>
                        {formatValue?.(item) ?? get(key, item)}
                    </TableTd>
                ))}
            <TableTd style={{ width: '35%' }}>
                <Group gap="md" justify="center">
                    <ActionIcon onClick={handleEdit} variant="subtle">
                        <EditPencil />
                    </ActionIcon>
                    <Popover
                        closeOnClickOutside
                        closeOnEscape
                        trapFocus
                        withArrow
                        withinPortal
                        id="delete-confirmation"
                        onChange={setHasOpenConfirmation}
                        opened={hasOpenConfirmation}
                        shadow="md"
                    >
                        <Popover.Target>
                            <ActionIcon
                                color="red"
                                onClick={toggleConfirmation}
                                variant="subtle"
                            >
                                <Trash />
                            </ActionIcon>
                        </Popover.Target>
                        <Popover.Dropdown>
                            <Stack align="center" gap="md">
                                <Text>{t('confirmation.generic')}</Text>
                                <Group gap="md">
                                    <Button
                                        color="red"
                                        onClick={toggleConfirmation}
                                        size="xs"
                                        variant="outline"
                                    >
                                        {t('confirmation.no')}
                                    </Button>
                                    <Button
                                        loading={deleteInProgress}
                                        onClick={handleDelete}
                                        size="xs"
                                    >
                                        {t('confirmation.yes')}
                                    </Button>
                                </Group>
                            </Stack>
                        </Popover.Dropdown>
                    </Popover>
                </Group>
            </TableTd>
        </TableTr>
    );
};
