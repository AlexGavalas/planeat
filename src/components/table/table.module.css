.table tr:nth-child(even) {
    background-color: #e5f6e5;
}
