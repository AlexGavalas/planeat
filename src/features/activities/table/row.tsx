import { Button, Group, Popover, Stack, Text } from '@mantine/core';
import { useModals } from '@mantine/modals';
import { showNotification } from '@mantine/notifications';
import { useSupabaseClient } from '@supabase/auth-helpers-react';
import { format, parseISO } from 'date-fns';
import { useTranslation } from 'next-i18next';
import { useState } from 'react';
import { useQueryClient } from 'react-query';

import { ActivityModal } from '~features/modals/activity';
import { useProfile } from '~hooks/use-profile';
import { type Activity } from '~types/activity';
import { type Database } from '~types/supabase';

interface RowProps {
    item: Activity;
    page: number;
}

export const Row = ({ item, page }: RowProps) => {
    const { t } = useTranslation();
    const modals = useModals();
    const supabase = useSupabaseClient<Database>();
    const queryClient = useQueryClient();
    const { profile } = useProfile();

    const [openConfirmation, setOpenConfirmation] = useState(false);
    const [deleteInProgress, setDeleteInProgress] = useState(false);

    const handleDelete = async () => {
        setDeleteInProgress(true);

        const { error } = await supabase
            .from('activities')
            .delete()
            .eq('id', item.id);

        if (error) {
            showNotification({
                title: t('error'),
                message: `${t('errors.activity_delete')}. ${t('try_again')}`,
                color: 'red',
            });
        } else {
            await queryClient.invalidateQueries(['activities-count']);
            await queryClient.invalidateQueries(['activities']);

            setDeleteInProgress(false);
            setOpenConfirmation(false);
        }
    };

    const handleSave = async () => {
        await queryClient.invalidateQueries(['activities-count']);
        await queryClient.invalidateQueries(['activities', page]);
    };

    return (
        <tr style={{ width: '100%', height: '3rem' }}>
            <td style={{ width: '25%' }}>
                {format(parseISO(item.date), 'dd/MM/yy')}
            </td>
            <td style={{ width: '40%' }}>{item.activity}</td>
            <td style={{ width: '35%' }}>
                <Group spacing="md" grow>
                    <Button
                        size="xs"
                        onClick={() => {
                            if (!profile) {
                                return;
                            }

                            modals.openModal({
                                title: t('add_activity'),
                                centered: true,
                                size: 'sm',
                                children: (
                                    <ActivityModal
                                        userId={profile.id}
                                        initialData={{
                                            id: item.id,
                                            date: parseISO(item.date),
                                            activity: item.activity,
                                        }}
                                        onSave={handleSave}
                                    />
                                ),
                            });
                        }}
                    >
                        {t('edit')}
                    </Button>
                    <Popover
                        shadow="md"
                        withArrow
                        withinPortal
                        trapFocus
                        closeOnEscape
                        closeOnClickOutside
                        opened={openConfirmation}
                        onChange={setOpenConfirmation}
                    >
                        <Popover.Target>
                            <Button
                                color="red"
                                size="xs"
                                onClick={() => setOpenConfirmation(true)}
                            >
                                {t('delete')}
                            </Button>
                        </Popover.Target>
                        <Popover.Dropdown>
                            <Stack spacing="md" align="center">
                                <Text>{t('confirmation.generic')}</Text>
                                <Group spacing="md">
                                    <Button
                                        color="red"
                                        size="xs"
                                        variant="outline"
                                        onClick={() => {
                                            setOpenConfirmation(false);
                                        }}
                                    >
                                        {t('confirmation.no')}
                                    </Button>
                                    <Button
                                        size="xs"
                                        onClick={handleDelete}
                                        loading={deleteInProgress}
                                    >
                                        {t('confirmation.yes')}
                                    </Button>
                                </Group>
                            </Stack>
                        </Popover.Dropdown>
                    </Popover>
                </Group>
            </td>
        </tr>
    );
};
