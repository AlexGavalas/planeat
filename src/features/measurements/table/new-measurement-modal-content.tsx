import { Button, Center, Group, NumberInput, Space, Text } from '@mantine/core';
import { DatePicker } from '@mantine/dates';
import { useModals } from '@mantine/modals';
import { showNotification } from '@mantine/notifications';
import { useSupabaseClient } from '@supabase/auth-helpers-react';
import 'dayjs/locale/el';
import { useTranslation } from 'next-i18next';
import { type FormEventHandler, useState } from 'react';

import { type Database } from '~types/supabase';

const localeMap = {
    gr: 'el',
};

interface ModalContentProps {
    userId: number;
    onSave: () => void;
    initialData?: {
        id: string;
        date: Date;
        weight?: number;
        fat_percentage?: number;
    };
}

export const NewMeasurementModalContent = ({
    userId,
    onSave,
    initialData,
}: ModalContentProps) => {
    const { t, i18n } = useTranslation();
    const supabaseClient = useSupabaseClient<Database>();
    const modals = useModals();
    const [date, setDate] = useState<Date | null>(
        initialData?.date ?? new Date(),
    );
    const [weight, setWeight] = useState(initialData?.weight);
    const [fatPercent, setFatPercent] = useState(initialData?.fat_percentage);
    const [error, setError] = useState('');

    const closeModal = () => modals.closeAll();

    const handleSave: FormEventHandler<HTMLFormElement> = async (e) => {
        e.preventDefault();

        if (!weight) return setError(t('errors.weight_empty'));
        if (!date) return setError(t('errors.date_empty'));

        let error;

        if (initialData) {
            const result = await supabaseClient
                .from('measurements')
                .update({
                    date: date.toISOString(),
                    weight,
                    fat_percentage: fatPercent,
                })
                .eq('id', initialData.id);

            error = result.error;
        } else {
            const result = await supabaseClient.from('measurements').insert({
                user_id: userId,
                date: date.toISOString(),
                weight,
                fat_percentage: fatPercent,
            });

            error = result.error;
        }

        if (error) {
            showNotification({
                title: t('error'),
                message: `${t('errors.measurement_save')}. ${t('try_again')}`,
                color: 'red',
            });
        } else {
            onSave?.();
            closeModal();
        }
    };

    return (
        <form onSubmit={handleSave} className="calendar">
            <Text>{t('date')}</Text>
            <Center>
                <DatePicker
                    value={date}
                    onChange={setDate}
                    locale={localeMap[i18n.language as keyof typeof localeMap]}
                />
            </Center>
            <Space h={20} />
            <NumberInput
                label={t('weight')}
                error={error}
                onFocus={() => setError('')}
                min={0}
                value={weight}
                onChange={(value) => setWeight(Number(value))}
                precision={2}
            />
            <Space h={20} />
            <NumberInput
                label={t('fat_label')}
                min={0}
                max={100}
                value={fatPercent}
                onChange={(value) => setFatPercent(Number(value))}
                precision={2}
            />
            <Space h={20} />
            <Group position="apart">
                <Button variant="light" color="red" onClick={closeModal}>
                    {t('cancel')}
                </Button>
                <Button type="submit" className="button">
                    {t('save')}
                </Button>
            </Group>
        </form>
    );
};
