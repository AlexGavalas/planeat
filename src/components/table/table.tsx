import {
    Group,
    Table as MantineTable,
    Pagination,
    Stack,
    TableTbody,
    TableTh,
    TableThead,
    TableTr,
} from '@mantine/core';
import { useState } from 'react';

import { type Header, type Item, Row } from './row';
import styles from './table.module.css';

export const INITIAL_PAGE = 1;
export const PAGE_SIZE = 5;

export type TableProps<ItemType> = {
    data: ItemType[];
    headers: Header<ItemType>[];
    totalPages: number;
    page: number;
    onDelete: (item: ItemType) => Promise<void> | void;
    onEdit: (item: ItemType) => Promise<void> | void;
    onPageChange?: (page: number) => void;
};

export const Table = <ItemType extends Item>({
    data,
    headers,
    totalPages,
    onPageChange,
    onDelete,
    onEdit,
    page: initialPage,
}: TableProps<ItemType>) => {
    const [page, setPage] = useState(initialPage);

    const shouldShowPagination = totalPages > 1;

    const handlePageChange = (page: number) => {
        setPage(page);
        onPageChange?.(page);
    };

    return (
        <Stack gap="md">
            <MantineTable className={styles.table}>
                <TableThead>
                    <TableTr>
                        {headers.map(({ key, label, width }) => (
                            <TableTh
                                key={label}
                                style={{
                                    width,
                                    textAlign:
                                        key === 'actions' ? 'center' : 'left',
                                }}
                            >
                                {label}
                            </TableTh>
                        ))}
                    </TableTr>
                </TableThead>
                <TableTbody>
                    {data.map((item) => (
                        <Row
                            key={item.id}
                            headers={headers}
                            item={item}
                            onDelete={onDelete}
                            onEdit={onEdit}
                        />
                    ))}
                </TableTbody>
            </MantineTable>
            {shouldShowPagination && (
                <Group justify="end">
                    <Pagination
                        withEdges
                        onChange={handlePageChange}
                        total={totalPages}
                        value={page}
                    />
                </Group>
            )}
        </Stack>
    );
};
