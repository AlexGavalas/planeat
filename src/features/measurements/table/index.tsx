import {
    ActionIcon,
    Card,
    Center,
    Group,
    Pagination,
    Stack,
    Table,
    Title,
} from '@mantine/core';
import { useModals } from '@mantine/modals';
import { useSupabaseClient } from '@supabase/auth-helpers-react';
import { Plus } from 'iconoir-react';
import { useTranslation } from 'next-i18next';
import { useState } from 'react';
import { useQuery, useQueryClient } from 'react-query';

import { LoadingOverlay } from '~components/loading-overlay';
import { MeasurementModal } from '~features/modals/measurement';
import { useProfile } from '~hooks/use-profile';
import { type Database } from '~types/supabase';

import { Row } from './row';

const PAGE_SIZE = 5;

export const MeasurementsTable = () => {
    const { t } = useTranslation();
    const modals = useModals();
    const { profile } = useProfile();
    const queryClient = useQueryClient();
    const supabase = useSupabaseClient<Database>();
    const [page, setPage] = useState(1);

    const { data: count = 0, isFetched } = useQuery(
        ['measurements-count'],
        async () => {
            if (!profile) {
                throw new Error(`User not logged in`);
            }

            const { count } = await supabase
                .from('measurements')
                .select('id', { count: 'exact' })
                .eq('user_id', profile.id);

            return count;
        },
        {
            enabled: Boolean(profile),
        },
    );

    const { data: measurements = [], isFetching } = useQuery(
        ['measurements', page],
        async () => {
            if (!profile) {
                throw new Error(`User not logged in`);
            }

            return supabase
                .from('measurements')
                .select('*')
                .eq('user_id', profile.id)
                .range((page - 1) * PAGE_SIZE, page * PAGE_SIZE - 1)
                .order('date', { ascending: false });
        },
        {
            select: ({ data }) => data || [],
            enabled: isFetched,
        },
    );

    const totalPages = Math.ceil((count || 0) / PAGE_SIZE);

    const onNewWeightSave = () => {
        setPage(1);
        queryClient.invalidateQueries(['measurements']);
    };

    const loading = (!measurements.length && !isFetched) || isFetching;

    return (
        <>
            <Group position="apart" py={20}>
                <Title order={3}>{t('measurements')}</Title>
                <ActionIcon
                    title={t('add_measurement')}
                    size="lg"
                    onClick={() => {
                        if (!profile) {
                            return;
                        }

                        modals.openModal({
                            title: t('new_measurement'),
                            centered: true,
                            size: 'sm',
                            children: (
                                <MeasurementModal
                                    userId={profile.id}
                                    onSave={onNewWeightSave}
                                />
                            ),
                        });
                    }}
                >
                    <Plus />
                </ActionIcon>
            </Group>
            <Card style={{ minHeight: 100 }}>
                <LoadingOverlay visible={loading} />
                {measurements.length > 0 ? (
                    <Stack spacing="md">
                        <Table highlightOnHover>
                            <thead>
                                <tr>
                                    <th style={{ width: '25%' }}>
                                        {t('date')}
                                    </th>
                                    <th style={{ width: '20%' }}>
                                        {t('weight')}
                                    </th>
                                    <th style={{ width: '20%' }}>
                                        {t('fat_label')}
                                    </th>
                                    <th style={{ width: '35%' }}>
                                        {t('actions')}
                                    </th>
                                </tr>
                            </thead>
                            <tbody>
                                {measurements.map((item) => (
                                    <Row
                                        key={item.id}
                                        item={item}
                                        page={page}
                                    />
                                ))}
                            </tbody>
                        </Table>
                        {totalPages > 1 && (
                            <Pagination
                                total={totalPages}
                                position="right"
                                value={page}
                                onChange={setPage}
                                withEdges
                            />
                        )}
                    </Stack>
                ) : (
                    !loading && (
                        <Center style={{ height: 100 }}>
                            <Title order={4}>{t('no_measurements_yet')}</Title>
                        </Center>
                    )
                )}
            </Card>
        </>
    );
};
