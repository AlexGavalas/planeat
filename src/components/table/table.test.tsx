import { renderWithUser } from '~test/utils';

import { Table, type TableProps } from './table';

describe('<Table />', () => {
    const onDelete = jest.fn();
    const onEdit = jest.fn();

    const props = {
        data: [{ id: 1, key: 'test value 1' }],
        headers: [{ key: 'key', label: 'test label', width: '100%' }],
        onDelete,
        onEdit,
        totalPages: 1,
        page: 1,
    } satisfies TableProps;

    describe('when passed no data', () => {
        it('renders a table', () => {
            const { container } = renderWithUser(
                <Table {...props} data={[]} headers={[]} totalPages={0} />,
            );

            expect(container).toMatchSnapshot();
        });
    });

    describe('when passed data', () => {
        it('renders a table', () => {
            const { container } = renderWithUser(<Table {...props} />);

            expect(container).toMatchSnapshot();
        });
    });
});
