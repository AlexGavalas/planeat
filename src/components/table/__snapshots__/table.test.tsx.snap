// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> when passed data renders a table 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m-6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-md);"
  >
    <table
      class="table m-b23fa0ef mantine-Table-table"
      style="--table-vertical-spacing: calc(0.4375rem * var(--mantine-scale));"
    >
      <thead
        class="mantine-Table-thead"
      >
        <tr
          class="m-4e7aa4fd mantine-Table-tr"
          data-with-row-border="true"
        >
          <th
            class="m-4e7aa4f3 mantine-Table-th"
            style="text-align: left; width: 100%;"
          >
            test label
          </th>
        </tr>
      </thead>
      <tbody
        class="m-b2404537 mantine-Table-tbody"
      >
        <tr
          class="m-4e7aa4fd mantine-Table-tr"
          data-with-row-border="true"
          style="height: 3rem; width: 100%;"
        >
          <td
            class="m-4e7aa4ef mantine-Table-td"
            style="width: 100%;"
          >
            test value 1
          </td>
          <td
            class="m-4e7aa4ef mantine-Table-td"
            style="width: 35%;"
          >
            <div
              class="m-4081bf90 mantine-Group-root"
              style="--group-gap: var(--mantine-spacing-md); --group-justify: center;"
            >
              <button
                class="mantine-focus-auto mantine-active m-8d3f4000 mantine-ActionIcon-root m-87cf2631 mantine-UnstyledButton-root"
                data-variant="subtle"
                style="--ai-bg: transparent; --ai-hover: var(--mantine-color-blue-light-hover); --ai-color: var(--mantine-color-blue-light-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
                type="button"
              >
                <svg
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.02 5.828L15.85 3l4.949 4.95-2.829 2.828m-4.95-4.95l-9.606 9.607a1 1 0 00-.293.707v4.536h4.536a1 1 0 00.707-.293l9.606-9.607m-4.95-4.95l4.95 4.95"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
              <button
                aria-controls="delete-confirmation-dropdown"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="mantine-focus-auto mantine-active m-8d3f4000 mantine-ActionIcon-root m-87cf2631 mantine-UnstyledButton-root"
                data-variant="subtle"
                id="delete-confirmation-target"
                style="--ai-bg: transparent; --ai-hover: var(--mantine-color-red-light-hover); --ai-color: var(--mantine-color-red-light-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
                type="button"
              >
                <svg
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 9l-1.995 11.346A2 2 0 0116.035 22h-8.07a2 2 0 01-1.97-1.654L4 9M21 6h-5.625M3 6h5.625m0 0V4a2 2 0 012-2h2.75a2 2 0 012 2v2m-6.75 0h6.75"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<Table /> when passed no data renders a table 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m-6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-md);"
  >
    <table
      class="table m-b23fa0ef mantine-Table-table"
      style="--table-vertical-spacing: calc(0.4375rem * var(--mantine-scale));"
    >
      <thead
        class="mantine-Table-thead"
      >
        <tr
          class="m-4e7aa4fd mantine-Table-tr"
          data-with-row-border="true"
        />
      </thead>
      <tbody
        class="m-b2404537 mantine-Table-tbody"
      />
    </table>
  </div>
</div>
`;
